src/org/folio/models/EurekaTenant.groovy

```diff
@@ -0,0 +1,53 @@
+package org.folio.models
+
+/**
+ * EurekaTenant class representing a tenant configuration for Eureka.
+ * It provides chainable setter methods following builder pattern for ease of use.
+ */
+class EurekaTenant extends Tenant {
+  /**
+   * Keycloak client identifier.
+   */
+  String clientId
+
+  /**
+   * Some secret phrase to authenticate in Keycloak.
+   * Is stored in AWS SSM Parameter Store or Hashicorp Vault.
+   */
+  String clientSecret
+
+  /**
+   * Keycloak service URL.
+   * Is the same for all Tenants
+   */
+  String keycloakUrl
+
+  /**
+   * Kong service URL.
+   * Is the same for all Tenants
+   */
+  String kongUrl
+
+  /**
+   * Tenant Manager Service URL.
+   * Is the same for all Tenants.
+   */
+  String tenantManagerUrl
+
+  /**
+   * Constructor that sets the tenantId and initializes modules.
+   * @param tenantId Tenant's identifier.
+   * @param clientId Keycloak client identifier.
+   * @param clientSecret Some secret phrase to authenticate in Keycloak.
+   * @param keycloakUrl Keycloak service URL.
+   * @param kongUrl Kong service URL.
+   */
+  EurekaTenant(String tenantId, String tenantDescription, String clientId, String clientSecret, String keycloakUrl, String kongUrl) {
+    super(tenantId, tenantDescription)
+    this.clientId = clientId
+    this.clientSecret = clientSecret
```

Comment:
Is clientSecret something that always known at class initialisation stage?

Reply:
@OHaimanov 

In the Future we may have every Application Module having its own **clientId** with dedicated secret password for getting its own access token.
That is why I put this attribute here for future usage.

Reply:
Without known **clientId** and **SecretPassword** application module will not be able to get valid access token from Keycloak.

Reply:
Ok, I mentioned these just to point that if this filed smth that will come after auth request or smth else,  that it is better to use it via setter, not during class initialization